### PR TITLE
Add support for custom elements

### PR DESCRIPTION
**Pull request by 50Wliu**
Copied from language-css.

/cc @esdoppio

Supersedes and closes #59

### CODE REVIEW

File: grammars/less.cson [hunk 1]
```diff
@@ -282,15 +282,15 @@
   }
   {
     'match': '\\b(a|abbr|acronym|address|applet|area|article|aside|audio|b|base|bdi|bdo|big|blockquote|body|br|button|canvas|caption|center|cite|code|col|colgroup|data|datalist|dd|del|details|dfn|dialog|div|dl|dt|em|embed|eventsource|fieldset|figure|figcaption|footer|form|frame|frameset|(h[1-6])|head|header|hgroup|hr|html|i|iframe|img|input|ins|kbd|keygen|label|legend|li|link|main|map|mark|math|menu|menuitem|meta|meter|nav|noframes|noscript|object|ol|optgroup|option|output|p|param|picture|pre|progress|q|rb|rp|rt|rtc|ruby|s|samp|script|section|select|small|source|span|strike|strong|style|sub|summary|sup|svg|table(?!-)|tbody|td|template|textarea|tfoot|th|thead|time|title|tr|track|tt|u|ul|var|video|wbr)\\b'
-    'name': 'keyword.control.html.elements'
+    'name': 'entity.name.tag.html.css'
```
Comment (esdoppio) [inline grammars/less.cson]: Also, do you think it's better to change the boundary assertions to `(?<!\\w-)` and `(?!\\w-)` so this won't mess up with the tokenization of, say, `article-title` (as the name of a custom element)?

File: grammars/less.cson [hunk 2]
```diff
@@ -282,15 +282,15 @@
...
   }
   {
     'match': '\\b(vkern|view|use|tspan|tref|title|textPath|text|symbol|switch|svg|style|stop|set|script|rect|radialGradient|polyline|polygon|pattern|path|mpath|missing-glyph|metadata|mask|marker|linearGradient|line|image|hkern|glyphRef|glyph|g|foreignObject|font-face-uri|font-face-src|font-face-name|font-face-format|font-face|font|filter|feTurbulence|feTile|feSpotLight|feSpecularLighting|fePointLight|feOffset|feMorphology|feMergeNode|feMerge|feImage|feGaussianBlur|feFuncR|feFuncG|feFuncB|feFuncA|feFlood|feDistantLight|feDisplacementMap|feDiffuseLighting|feConvolveMatrix|feComposite|feComponentTransfer|feColorMatrix|feBlend|ellipse|desc|defs|cursor|color-profile|clipPath|circle|animateTransform|animateMotion|animateColor|animate|altGlyphItem|altGlyphDef|altGlyph|a)\\b'
-    'name': 'keyword.control.html.svg.elements'
+    'name': 'entity.name.tag.html.svg.css'
```
Comment (esdoppio) [inline grammars/less.cson]: Also, you could remove the `title`, `style`, `script` here, coz they won't be matched by this rule.

**Comment by 50Wliu** [comment]
Sure.  I was trying to stay close to the existing pattern names, but it would be better to have consistency across the board.

File: grammars/less.cson (hunk 1 above)
Comment (Alhadis) [inline grammars/less.cson]: +1 to that. `\b` === `\bad`

**Comment by 50Wliu** [comment]
Review comments have been addressed.  I also made the regex extended for easier readability and shortened it a bit.

File: grammars/less.cson [hunk 3]
```diff
@@ -281,12 +281,41 @@
...
+      |canvas|caption|center|cite|code|col|colgroup
+      |data|datalist|dd|del|details|dfn|dialog|div|dl|dt
+      |em|embed|eventsource|fieldset|figure|figcaption|footer|form|frame|frameset
+      |h[1-6]|head|header|hgroup|hr|html|i|iframe|img|input|ins
+      |kbd|keygen|label|legend|li|link|main|map|mark|math|menu|menuitem|meta|meter
+      |nav|noframes|noscript|object|ol|optgroup|option|output
+      |p|param|picture|pre|progress|q|rb|rp|rt|rtc|ruby
+      |s|samp|script|section|select|small|source|span|strike|strong|style|sub|summary|sup|svg
+      |table|tbody|td|template|textarea|tfoot|th|thead|time|title|tr|track|tt
+      |u|ul|var|video|wbr)
```
Comment (esdoppio) [inline grammars/less.cson]: You could remove acronym, applet, big, center, eventsource, frame, frameset, noframes, strike, and tt.

Comment (50Wliu) [inline grammars/less.cson]: Done.

**Pull request merged**